bfvmm/integration/arch/intel_x64/test_all.sh

```diff
@@ -0,0 +1,370 @@
+    local direction_count=$(grep "\- direction_of_access" serial.out | wc -l)
+    local address_count=$(grep "\- address " serial.out | wc -l)
+    local val_count=$(grep "\- val" serial.out | wc -l)
+
+    if [[ $record_count -eq $port_count && $port_count -eq $size_count &&
+          $size_count -eq $direction_count && $direction_count -eq $address_count &&
+          $address_count -eq $val_count ]];
+    then
+        return 0
+    fi
+
+    echo ""
+    echo_fail "observed $record_count records; expected $record_count"
+    echo_fail "observed $port_count port; expected $record_count"
+    echo_fail "observed $size_count size; expected $record_count"
+    echo_fail "observed $direction_count direction; expected $record_count"
+    echo_fail "observed $address_count address; expected $record_count"
+    echo_fail "observed $val_count val; expected $record_count"
+    echo ""
+
+    exit 1
+}
+
+check_test()
+{
+    local vmm=$1
+
+    case $vmm in
+        *_x64_drs*) check_dr;;
+        *_x64_cpuid*) check_cpuid;;
+        *_x64_control_register*) check_cr;;
+        *_x64_io_instruction*) check_io;;
+        *_x64_monitor_trap*) check_monitor_trap;;
+        *_x64_*msr*) check_msr;;
+        *_x64_vic*) check_vic;;
+        *_x64_vpid*) check_vpid;;
+        *)
+            echo ""
+            echo -ne "$bold_yellow"
+            echo -e  "ALERT:$reset check not supported: $vmm"
+            ;;
+    esac
+}
+
+#
+# Test initializers
+#
+
+init_vic()
+{
+    sudo dmesg -C
+}
+
+init_test()
+{
+    local vmm=$1
+
+    case $vmm in
+        *_x64_vic*) init_vic;;
+        *) ;;
+    esac
+}
+
+#
+# Prettify the test names
+#
+
+parse_test_name()
+{
+    name=$(basename $1 | sed 's|eapis_integration_intel_x64_\(.*\)_static|\1|')
+    echo "$name"
+}
+
+#
+# Global variables
+#
+
+case $LANG in
+    en_US\.UTF-8)
+        pass_mark='\u2713'
+        fail_mark='\u2718'
+        ;;
+    *)
+        pass_mark='*'
+        fail_mark='x'
+        ;;
+esac
+
+pass_mark="$bold_green$pass_mark$reset"
+fail_mark="$bold_red$fail_mark$reset"
+
+build_dir=$HOME/bareflank/build-eapis
+vmm_bin_dir=$build_dir/prefixes/x86_64-vmm-elf/bin
+
+cpu_count=$(nproc)
+test_list=$(find $vmm_bin_dir -name "*eapis*integration*" | sort)
+bfm=$build_dir/prefixes/x86_64-userspace-elf/bin/bfm
+
+#
+# Test code
+#
+
+echo_milestone "Preparing test vmms\n"
+echo_task "building test vmms\n"
+cmake --build $build_dir > /dev/null
+
+echo_task "cleaning bfdriver\n"
+cmake --build $build_dir --target driver_clean > /dev/null
+
+echo_task "building bfdriver\n"
+cmake --build $build_dir --target driver_build > /dev/null
+
+echo_task "loading  bfdriver\n"
+cmake --build $build_dir --target driver_load > /dev/null
+
+echo_milestone "Running test vmms\n"
+for vmm in $test_list
+do
+    echo_task "testing: $(parse_test_name $vmm)"
+    init_test $vmm
+
+    sudo $bfm load $vmm
+    sudo $bfm start
+    sleep 1
+    sudo $bfm stop
+    sudo $bfm dump > serial.out
+
+    check_test $vmm
```

Comment:
On my environment, the first integration test fails and the tests stop. Just checking that is the intended behavior?

Reply:
New version should work now. The NDEBUG flag has to be not defined or else the output that the verifiers parse wont be printed to serial. Also the script takes an argument to the cmake build directory, hypervisor source, and a cmake config file. The config file should enable the eapis extension to build the integration vmms